src/frequenz/channels/_merge.py

```diff
@@ -53,7 +56,13 @@ def __init__(self, *receivers: Receiver[_T], name: str | None) -> None:
             *receivers: The receivers to merge.
             name: The name of the receiver. Used to to create the string representation
                 of the receiver.
+
+        Raises:
+            ValueError: if less than two receivers are provided.
         """
+        if len(receivers) < 2:
+            raise ValueError("At least two receivers must be provided")
```

Comment:
if it is a single component, we could actually just return the receiver, instead of a `_Merge`.  and raise only when there are 0 receivers.

that way, users can still say "get receivers for all batteries and  merge them" and have to do special case handling only for one case.

Reply:
I agree, maybe all we need are some logs instead of exceptions.

Reply:
We could have a c'tor flag or parameter `allow_empty=` or `min_required_receivers =`

Reply:
I'm also not convinced about the logging, as if you call `merge()` with no receivers (or one) on purpose, then you'll get your logs littered with basically noise. We could add those ctor flags to decide if we raise or emit logs but I find it overkill to be honest.

My feeling for now is:

* `merge(receiver)` should be fine, there are really no surprises there, raising or logging here is being overly patronizing, as everything will work as expected
* `merge()` should raise. It needs special casing anyway, as it would be very weird that one wants the `merge()` loop to exit without doing any iteration, and if that's your intention, then it can be explicitly expressed using:

  ```python
  receivers = []
  if x:
      receivers.add(x)
  if y:
      receivers.add(x)
  if receivers:
      async for m in merge(*receivers):
          ...
  ```

  Which is only one extra line and makes everything very clear and explicit.

Reply:
OK, I went with this approach (raise with no receivers, optimize to return the same receiver if only one is provided). I also moved the raising to the `merge()` function, as `_Merge` is a low level construct. This way I can also avoid the `# noqa` for the indirectly raised exception.